.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,125 @@
+
+            - name: Create Pull Request
+              env:
+                  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              run: |
+                  cat > pr_body.txt << 'EOF'
+                  ## Release v${{ inputs.version }}
+
+                  This PR prepares the release for version **${{ inputs.version }}**.
+
+                  ### Release Checklist
+                  - [x] Version set to ${{ inputs.version }}
+                  - [ ] Fix any deprecation deadlines if they exist
+                  - [ ] Integration tests pass (tagged with `integration-tests`)
+                  - [ ] Example checks pass (tagged with `test-examples`)
+                  - [ ] Draft release created at https://github.com/OpenHands/software-agent-sdk/releases/new
+                    - [ ] Select tag: `v${{ inputs.version }}`
+                    - [ ] Select branch: `${{ env.BRANCH_NAME }}`
+                    - [ ] Auto-generate release notes
+                    - [ ] Publish release (PyPI will auto-publish)
+                  - [ ] Evaluation on OpenHands Index
+
+                  ### Next Steps
+                  1. Review the version changes
+                  2. Address any deprecation deadlines
+                  3. Ensure integration tests pass
+                  4. Ensure example tests pass
+                  5. Create and publish the release
+
+                  Once the release is published on GitHub, the PyPI packages will be automatically published via the `pypi-release.yml` workflow.
+                  EOF
+
+                  gh pr create \
+                    --title "Release v${{ inputs.version }}" \
+                    --body-file pr_body.txt \
+                    --base main \
+                    --head "${{ env.BRANCH_NAME }}" \
+                    --label "integration-tests" \
+                    --label "test-examples"
+
+                  rm pr_body.txt
+                  echo "✅ Pull request created successfully!"
+
+                  # Get PR URL and display it
+                  PR_URL=$(gh pr view "${{ env.BRANCH_NAME }}" --json url --jq '.url')
+                  echo "🔗 PR URL: $PR_URL"
+                  echo "PR_URL=$PR_URL" >> $GITHUB_ENV
+
+            - name: Summary
+              run: |
+                  echo "## ✅ Release Preparation Complete!" >> $GITHUB_STEP_SUMMARY
+                  echo "" >> $GITHUB_STEP_SUMMARY
+                  echo "- **Version**: ${{ inputs.version }}" >> $GITHUB_STEP_SUMMARY
+                  echo "- **Branch**: ${{ env.BRANCH_NAME }}" >> $GITHUB_STEP_SUMMARY
+                  echo "- **PR URL**: ${{ env.PR_URL }}" >> $GITHUB_STEP_SUMMARY
+                  echo "" >> $GITHUB_STEP_SUMMARY
+                  echo "### Next Steps:" >> $GITHUB_STEP_SUMMARY
+                  echo "1. Review the PR and address any deprecation deadlines" >> $GITHUB_STEP_SUMMARY
+                  echo "2. Wait for integration tests and example checks to pass" >> $GITHUB_STEP_SUMMARY
```

Comment:
Will integration tests running automatically on release branch or does a label need to be assigned on the PR?